0.35

fabaff opened the pull request:
5000 stars on GitHub, 2000 people in out Gitter chatroom and over a million monthly page views. I don't think we could wish for a better place to be at the end of 2016. Feels like an early Christmas present! Our early one for you is 0.35. It's not a single thing inside a nice wrapping, more like several little gifts inside the 0.35 box.

This will be the last release of 2016 as our developers are taking a well deserved break. We will be back in 2017!

## Text to Speech

With the addition of a [text-to-speech][tts] component by [@pvizeli] we have been able to bring Home Assistant to a whole new level. The text-to-speech component will take in any text and will play it on a media player that supports to play media. We have tested this on Sonos, Chromecast, and Google Home.

https://www.youtube.com/watch?v=Ke0QuoJ4tRM

## Call for help with HASSbian (our Raspberry Pi image)

In an effort to make Home Assistant, we're planning to extend the things that people can do out of the box with HASSbian, our Raspberry Pi image. As you might know, the image is currently maintained by [@Landrash]. However he also spends a lot of time on improving the docs and helping out with a ton of other things.

So if you know your Linux-foo and would love to contribute to open source, join the [Hassbian developer chat](https://gitter.im/home-assistant/hassbian) and take a stab at one of [our open issues](https://github.com/home-assistant/pi-gen/issues).

## VLC media player, Broadlink, and GPSLogger
Let you control [VLC media player][vlc-media] and  play sound on connected speakers.

This Broadlink [switch][bl-switch] platform allow to you control Broadlink RM2 Pro and RM mini IR+RF devices. This Broadlink [sensor][bl-sensor] platform adds support for Broadlink RM2 and A1 Devices.

The [GPSLogger](https://home-assistant.io/components/device_tracker.gpslogger/) now also supports attributes as the speed, direction, altitude, provider and activity.

## All changes
- Media player: Support for Sharp [AquaosTV][aquostv] ([@titilambert])
- Media player: [VLC][vlc-media] support ([@Danielhiversen])
- Switch: Digital Loggers relay([@dethpickle])
- Sensor: Support for [Netdata][netdata] ([@fabaff], [@ezar])
- Sensor: Support weather conditions from Austrian [ZAMG][zamg] ([@mjl])
- Verisure: Add Verisure smartcam capture service ([@turbokongen])
- Binary sensor: [Flic][flic] button support added ([@soldag])
- Sensor: Support for [SenseHat][sensehat] ([@farminf])
- Binary sensor: [Hikvision][hikvision] binary sensor support ([@mezz64])
- [Text-to-speech][tts] support ([@pvizeli])
- Sensor: Support for Broadlink [sensors][bl-sensor] ([@Danielhiversen])
- Switch: Broadlink [switches][bl-switch] supported now ([@Danielhiversen])

- Media player: Add `source_list` to universal media player ([@danieljkemp])
- Binary Sensor: Support improvement for Wink ([@w1ll1am23])
- Sensor: More features for DSMR ([@aequitas])
- Automation: Parse payload as JSON ([@balloob])
- Mediap player - Cast: New progress indicator ([@balloob])
- Device tracker: New attributes ([@balloob], [@Danielhiversen])
- Binary sensor - netatmo: Add support for tags ([@jabesq])
- Climate: Add `away_mode` for RadioThermostat ([@trisk])
- Device tracker - nmap: Make scan option configurable ([@Danielhiversen])
- Climate - Ecobee: Add SmartAway option ([@devdelay])
- Light - Hue: Add support for Hue LightGroups ([@michaelarnauts])
- Media player - Emby: New support for trailer and media position ([@mezz64])
- Camera - Amcrest: Support for resolution ([@tchellomello])
- Media player - Kodi: Authentication fixed and fan art ([@balloob], [@joopert])
- Minor and not so minor features and bug fixes by [@pvizeli], [@jminn], [@magicus], [@teodoc], [@fabaff], [@technicalpickles], [@balloob], [@lukas-hetzenecker], [@rubund], [@dasos], [@trisk], [@armills], [@auduny], [@lwis], [@nkgilley], [@janLo], [@keatontaylor], [@stefan-jonasson], [@Jypy], [@jawilson], [@DavidLP], [@molobrakos], [@jabesq], [@joerocklin], [@kellerza], [@kirichkov], and [@danielperna84].

### If you need help...
...don't hesitate to use our [Forum](https://community.home-assistant.io/) or join us for a little [chat](https://gitter.im/home-assistant/home-assistant). The release notes have comments enabled but it's preferred if you use the former communication channels. Thanks.

### Reporting Issues
Experiencing issues introduced by this release? Please report them in our [issue tracker](https://github.com/home-assistant/home-assistant/issues). Make sure to fill in all fields of the issue template.

[@aequitas]: https://github.com/aequitas
[@armills]: https://github.com/armills
[@auduny]: https://github.com/auduny
[@balloob]: https://github.com/balloob
[@Danielhiversen]: https://github.com/Danielhiversen
[@danieljkemp]: https://github.com/danieljkemp
[@danielperna84]: https://github.com/danielperna84
[@dasos]: https://github.com/dasos
[@DavidLP]: https://github.com/DavidLP
[@dethpickle]: https://github.com/dethpickle
[@devdelay]: https://github.com/devdelay
[@ezar]: https://github.com/ezar
[@fabaff]: https://github.com/fabaff
[@farminf]: https://github.com/farminf
[@jabesq]: https://github.com/jabesq
[@janLo]: https://github.com/janLo
[@janLo]: https://github.com/janLo
[@jawilson]: https://github.com/jawilson
[@jawilson]: https://github.com/jawilson
[@jminn]: https://github.com/jminn
[@joerocklin]: https://github.com/joerocklin
[@joopert]: https://github.com/joopert
[@Jypy]: https://github.com/Jypy
[@keatontaylor]: https://github.com/keatontaylor
[@kellerza]: https://github.com/kellerza
[@kirichkov]: https://github.com/kirichkov
[@Landrash]: https://github.com/Landrash
[@lukas-hetzenecker]: https://github.com/lukas-hetzenecker
[@lwis]: https://github.com/lwis
[@magicus]: https://github.com/magicus
[@MartinHjelmare]: https://github.com/MartinHjelmare
[@mezz64]: https://github.com/mezz64
[@michaelarnauts]: https://github.com/michaelarnauts
[@mjl]: https://github.com/mjl
[@molobrakos]: https://github.com/molobrakos
[@nkgilley]: https://github.com/nkgilley
[@pvizeli]: https://github.com/pvizeli
[@rubund]: https://github.com/rubund
[@rubund]: https://github.com/rubund
[@soldag]: https://github.com/soldag
[@stefan-jonasson]: https://github.com/stefan-jonasson
[@tchellomello]: https://github.com/tchellomello
[@technicalpickles]: https://github.com/technicalpickles
[@teodoc]: https://github.com/teodoc
[@titilambert]: https://github.com/titilambert
[@trisk]: https://github.com/trisk
[@turbokongen]: https://github.com/turbokongen
[@w1ll1am23]: https://github.com/w1ll1am23

[vlc-media]: https://home-assistant.io/components/media_player.vlc/
[aquostv]: https://home-assistant.io/components/media_player.aquostv/
[digitalloggers]: https://home-assistant.io/components/switch.digitalloggers/
[netdata]: https://home-assistant.io/components/switch.digitalloggers/
[bl-sensor]: https://home-assistant.io/components/sensor.broadlink/
[bl-switch]: https://home-assistant.io/components/switch.broadlink/
[hikvision]: https://home-assistant.io/components/binary_sensor.hikvision/
[zamg]: https://home-assistant.io/components/sensor.zamg/
[flic]: https://home-assistant.io/components/binary_sensor.flic/
[sensehat]: https://home-assistant.io/components/sensor.sensehat/
[tts]: https://home-assistant.io/components/tts/

balloob committed:
Version bump to 0.35.0dev0

Commit:
device tracker

pvizeli committed:
Bugfix sonos hosts (#4698)

jminn committed:
Updated email message headers to have 'Date' and 'Message-Id' fields (#4693) (#4695)

magicus committed:
Remove global variable from tellstick code (#4700)

* Refactor tellstick code for increased readability. Especially highlight if "device" is a telldus core device or a HA entity.

* Refactor Tellstick object model for increased clarity.

* Update comments. Unify better with sensors. Fix typo bug. Add debug logging.

* Refactor tellstick code for increased readability. Especially highlight if "device" is a telldus core device or a HA entity.

* Refactor Tellstick object model for increased clarity.

* Update comments. Unify better with sensors. Fix typo bug. Add debug logging.

* Fix lint issues.

* Remove global variable according to hint from balloob.

teodoc committed:
Add option to hide the group card switch (#4631)

* Add option to hide the group card switch

* Disallow control of hidden group switches

* Revert "Disallow control of hidden group switches"

This reverts commit 75e5ddfe3092327647e2873cb03ccf3b4a75b85a.

* Changed hide_switch to control

technicalpickles committed:
Updated python-nest to fix a camera bug when loading images (#4701)

pvizeli committed:
Migrate weather to async (#4677)

especially-relative committed:
Universal source list (#4086)

* Add source_list to universal media player

* Expanded attirubte and command support for UMP

Added support to the universal media player
for the following:
    Volume Set
    Current Source
    Set Source
    Current Volume

The goal is to facilitate a single-card media player
that includes source selection and setting the volume
of the receiver.

Example setup:
```
media_player:
  - platform: universal
    name: Media Center
    children:
      - media_player.kodi
      - media_player.cast
    commands:
      select_source:
        service: media_player.select_source
        data:
          entity_id: media_player.receiver
      volume_set:
        service: media_player.volume_set
        data:
          entity_id: media_player.receiver
      volume_mute:
        service: media_player.volume_mute
        data:
          entity_id: media_player.receiver
      turn_on:
        service: homeassistant.turn_on
        data:
          entity_id: media_player.receiver
      turn_off:
        service: homeassistant.turn_off
        data:
          entity_id: media_player.receiver
    attributes:
      state: media_player.receiver
      is_volume_muted: media_player.receiver|is_volume_muted
      volume_level: media_player.receiver|volume_level
      source: media_player.receiver|source
      source_list: media_player.receiver|source_list
```

* Remove print statements

* Change service call back to use call_from_config

* Modified service calls to use template data

* linting fixes

* Add tests

* linting fices

* More pylinting

aequitas committed:
Dsmr hourly gas usage. (#4609)

* Hourly rate of Gas consumption. Use proper unknown state.

* Import unknown state constant.

* doh

* Cleanup device add.

* Fix lint.

* Add test for derivative calculation.

* Remove conflict.

* Document and move calculation into update call.

Commit:
Wink hub sensor (#4704)

pvizeli committed:
Migrate remote to async (#4678)

* Migrate remote to async

* add coro

* remove sync from init since only used in harmony

* import ATTR from remote

* remove unused sync stuff from tests

titilambert committed:
Add Sharp AquosTV component (#4679)

tchellomello committed:
Added persistent_notification in case of error during Unifi device_tracker setup (#4682)

dethpickle committed:
New support Digital Loggers relays (#4684)

* initial commit

Previous work included with no history.  Sorry, I was figuring out how to use git, branches and deal with open source projects.  At this point this is a working switch but with the shortcomings of each of the 8 ports causes a network query.  This needs to be rewritten so that the SwitchDevice is part of a larger device group that is only queried once, saving traffic and preventing the small device from timing out.

* Device polls independent of switches now

Used anel_pwrctrl.py as a basis to extract the per-switch polling out to per-device so it can be trottled properly.  Likewise, no longer touching the device independently for relay status AND relay name.  Getting them both from the same statuslist() return.

* Final comments and tweaks

Lowered cycle and update time since the device update is working so well now.  Effectively no timeouts anymore.

* Added dlipower to requirements

homeassistant.components.switch.digitalloggers

* Tox fixes

pydocstyle updates

* More tox errors

* Yet more tox

Removed useful future TODO and helpful details on the structure of the statuslocal list.
Good catch on not initializing .update(), though it worked.

* Blank line fix

* Added file to .coveragerc

pvizeli committed:
Protect hm thread for hangs on events (#4717)

mweinelt committed:
Revert "Update reference to correct tplink switch" (#4722)

aequitas committed:
Actually test calling async macvendor lookup and fix it. (#4718)

balloob committed:
MQTT Automation: parse payload as JSON (#4703)

balloob committed:
Fix synology dsm doing I/O inside loop (#4699)

balloob committed:
Fix CORS when static resources registered (#4727)

balloob committed:
Re-org emulated_hue and fix google home (#4708)

balloob committed:
Update netdisco to 0.8 (#4723)

balloob committed:
Fix Nest interpreting Celsius temperature as Fahrenheit (#4729)

technicalpickles committed:
Make sure all nest platforms require discovery info (#4734)

lukas-hetzenecker committed:
[0.34] bugfix influxdb node_id (#4712)

* Bugfix for #4709 - do not convert node_id to float

* Update influxdb.py

balloob committed:
Merge branch 'release-0-34-2' into dev

balloob committed:
Cast progress (#4735)

* add progress to google cast

* Add progress to media player demo

fabaff committed:
Netdata sensor (#4743)

* Added netdata sensor

* Typo

* Add netdata sensor

* Improvement of the work done by @ezar

Danielhiversen committed:
Merge pull request #4719 from home-assistant/device_tracker

device tracker

rubund committed:
Fix broken EnOcean support (#4710)

* ensure_list

* CONF_ID is not required configuration for enocean lights

* Use vol.All(cv.ensure_list, [vol.Coerce(int)]) as suggested in pull request review

* Fix line too long

balloob committed:
Device tracker attributes (#4753)

Danielhiversen committed:
Add additional attributes to GPSLogger (#4755)

jawilson committed:
Set hue-bridgeid in UPNP response (#4740)

Commit:
Add sensor for reading Austrian ZAMG weather conditions (#4347)

* Add sensor for reading ZAMG weather conditions

* Add to coveragerc; Correct some doc style problems

* More doc fixes

* More doc fixes

* Lose license and whatever.

* Don't return UNKNOWN for unknown variables

* Verify that the configured station id is actually one in the data set.

Don't warn about unknown stations, this cannot happen any more as the configuration parser now checks that.
This could still happen if the data set is incomplete though ...

* Clean up imports

* Clarify comment on throttling interval

* Base zamg sensor on Entity, not WeatherEntity, and delete unused code

* Fix formatting nits from flake8

* Use ATTR_FRIENDLY_NAME, clean up imports, remove unnecessary indirection.

* Use {}.format() instead of "" %

* Re-add unit of measurement that got lost somehow

* Use guard clauses instead of if-matroshka.
Wrap requests.get() in try/except for RequestException.

* Huh, how did this happen? White space corrections...

* Add sensor for reading ZAMG weather conditions

* Add to coveragerc; Correct some doc style problems

* More doc fixes

* More doc fixes

* Verify that the configured station id is actually one in the data set.

Don't warn about unknown stations, this cannot happen any more as the configuration parser now checks that.
This could still happen if the data set is incomplete though ...

* Lose license and whatever.

* Don't return UNKNOWN for unknown variables

* Clean up imports

* Clarify comment on throttling interval

* Base zamg sensor on Entity, not WeatherEntity, and delete unused code

* Fix formatting nits from flake8

* Use ATTR_FRIENDLY_NAME, clean up imports, remove unnecessary indirection.

* Use {}.format() instead of "" %

* Re-add unit of measurement that got lost somehow

* Use guard clauses instead of if-matroshka.
Wrap requests.get() in try/except for RequestException.

* Huh, how did this happen? White space corrections...

* Precipitation actually is a float, good it rained today

* Logger needs no module visibility

* Do not name sensors with _ to be in line with the other weather sensor platforms.

* Remove manually set friendly_name

* comment format police

* Less comments

* Update zamg.py

turbokongen committed:
Add Verisure smartcam capture service (#4559)

* Add verisure capture as service

* docstyle

balloob committed:
Fix websocket async (#4752)

* Ensure we write to websocket from inside event loop

* Inline service call helper

Danielhiversen committed:
fix setting battery in device_tracker (#4756)

emlove committed:
Fix media_image_urls for universal media player (#4765)

* Fix media_image_urls for universal media player

* Linter fixes

soldag committed:
Add flic smart button component (#4681)

* Added component for flic smart buttons.

* Apply home-assistant coding styles.

* Fixed flic configuration.

* Made logging for scanning for new buttons less verbose.

* Fixed flic event data.

* Follow async conventions.

* Added new requirements to requirements_all.txt.

* Added flic component to .coveragerc

* Updated flic threshold configuration key names.

* Flic devices are now removed when they disconnect.

* Include review feedback.

* Fixed stopping of clients in flic component when home assistant is stopped.

* Updated flic component by integrating input of #4738.

Use library method to determine click type. Merge three click events into single one with click_type parameter.

* Use a single client for both handling click events and scanning for new buttons.

* Renamed flic ‘auto_scan’ configuration variable to ‘discovery’ using HA constants.

jabesq committed:
Add support for Netatmo tags (#4761)

* Add support for Netatmo Welcome Tags

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Add size parameter for WelcomeData

* minor fixes

* Add Throttling mechanism for update event

This will prevent to reach the API limit

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Change scan interval for Netatmo Binary sensors

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Minor fixes

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Update netatmo.py

finish06 committed:
Pyunifi dep (#4754)

* change unifi dependency to pyunifi

* Change dependency to fix #4336

* Run gen_requirements_all.py script

* Changed import statement to reflect new package

* Updated test_unifiy.py with different module

* Update requirements_all.txt

dasos committed:
Fix connection check (#4732)

* Fix connection check

* Release instead

* Remove if

* Update hook.py

balloob committed:
Update frontend

trisk committed:
Match uppercase MAC addresses in asuswrt 'arp -n' output (#4742) (#4764)

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Fix default auth influxdb (#4771)

lwis committed:
Add test to ensure device_tracker records state correctly. (#4776)

fabaff committed:
Move details to docs, update doc strings, and use consts (#4777)

balloob committed:
Fix Kodi auth (#4770)

auduny committed:
Add support for telldus in the Docker image.  (#4680)

* Add support for telldus in the Docker image. Start with -v /tmp/TelldusClient:/tmp/TelldusClient -v /tmp/TelldusEvents:/tmp/TelldusEvents

* Merged telldus install with the others

* Clean up indenting

* Stream apt-key

technicalpickles committed:
Updated python-nest dependency (#4785)

* Updated python-nest dependency

This sha fixes two issues:

- min and max temperatures not being set when temperature isn't locked
- fixes error when setting farenheit with a .5

* gen requirements all

* Add fix for https://github.com/home-assistant/home-assistant/issues/4731

emlove committed:
Fix incorrect caching of /api/error_log (#4789)

balloob committed:
Add faster reviews link to contributing

balloob committed:
Contributing: add skip step 0

balloob committed:
Update after service calls (#4795)

* Update after service calls

* Service update: wrap async_update in create_task

nkgilley committed:
round $ to 2 decimals (#4786)

janLo committed:
Scene: add support for input_select (#4674)

This adds support for the scene component to handle input_select
devices and set their options.

This fixes bug #4673

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

pvizeli committed:
Pararell execute state restore by domain (#4801)

* Pararell execute state restore per domain

* fix spell

keatontaylor committed:
Get entity name from entity.name (#4798)

Grabbing the ATTR_FRIENDLY_NAME directly produces an error. Instead
grab from entity.name.

Commit:
Fix python-nest version bump (#4799)

* Fix python-nest version bump

* Change SHA so version in HASS and dependency are the same

trisk committed:
Add exception handling when turning on Onkyo receivers (#4813)

lwis committed:
[media_player.sonos] Add stop support. (#4788)

balloob committed:
Update frontend

janLo committed:
Pilight receive match fix for bug 4637 (#4639)

* Pilight: dont protocol as list in COMMAND_SCHEMA

As described in bug #4637 the protocol should not be wrapped in a list
in the spec of COMMAND_SCHEMA because this causes the component to
never successfully match any received rf code.

As pointed ot in PR #4639 the easiest way to do this, is to not derive
COMMAND_SCHEMA from RF_CODE_SCHEMA and specify protocol as simple
string there.

This fixes bug #4637.

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

* Pilight: Add "unitcode" to command schema.

This adds "unitcode" to the COMMAND_SCHEMA. It is used for example in
the brennenstuhl protocol of pilight.

Signed-off-by: Jan Losinski <losinski@wh2.tu-dresden.de>

pvizeli committed:
Bugfix error on automation reload (#4823)

trisk committed:
Add away mode for Radio Thermostat/3M Filtrete (#4793)

keatontaylor committed:
Prevent emulated hue discovery by hue component (#4819)

* Prevent emulated hue discovery

Test for “HASS Bridge” in discovery info, pass if found, else try and
setup the bridge.

* Solved coding error

Duplicate commands and return false added for component.

soldag committed:
Flic: Support ignoring individual click types. (#4827)

technicalpickles committed:
Bump python-nest to fix issue with Nest Cam without activity zones (#4820)

* Bump python-nest to fix issue with Nest Cam without activity zones

* bump to include fix python-nest dependency with hvac_state

* regenerate requirements_all.txt

r-jordan committed:
[climate.generic_thermostat] Make tolerance work both ways (#4830)

Danielhiversen committed:
Configurable scan options for nmap (#4838)

stefan-jonasson committed:
Fixed crash during light objects initizilation (#4835)

* Fixed crash when lights objects was inited

lwis committed:
[device_tracker.gpslogger] Add additional activity attribute.

Danielhiversen committed:
Merge pull request #4846 from lwis/gpslogger

[device_tracker.gpslogger] Add additional activity attribute.

lwis committed:
[device_tracker] Don't clear GPS coordinates if no GPS seen (#4848)

kellerza committed:
Revert "[device_tracker] Don't clear GPS coordinates when using two device trackers." (#4851)

pvizeli committed:
Migrate alarm control panel to async (#4807)

* Merge alarm control panel to async

* fix lint

dcrypt3d committed:
Ecobee detect Smart Away (#4769)

* Ecobee autoAway Event

* Update ecobee.py

Checking if event['running'] true is pointless because if false event['type'] will equal template and when true type will only be 'hold' or 'autoAway' so I've removed this check from the statement

Jypy committed:
Netatmo improving Battery info (#4724)

* Improving Battery info

Improving battery status info (to reflect NetAtmo API documentation and change deprecated DeviceList for WeatherStationData

* Fixes from previous update

Fix the hound issue.
add battery_lvl, WindAngle_value, GustAngle_value, rf_status_lvl, wifi_status_lvl

Danielhiversen committed:
vlc media player (#4800)

* vlc media player

* Update vlc.py

tchellomello committed:
Added support  to language codes on Weather Underground (#4815)

* Added supported to language codes to Weather Underground

* Removed unecessary None assigments

Commit:
adding sensehat plugin (#4775)

* adding sensehat plugin

* added

* fix PR

* requirement updated

* Update sensehat.py

jawilson committed:
Use current mode to determine which temperature attributes to use (#4858)

balloob committed:
Fix Plex from doing I/O inside event loop (#4857)

balloob committed:
Fix Nest doing I/O inside event loop (#4855)

DavidLP committed:
Fix config validation (#4853) (#4854)

michaelarnauts committed:
Add support for Hue LightGroups (#4744)

* Add support for Hue LightGroup entity

* Don't filter on LightGroup and add properties for a group

* Reuse code from HueLight in HueLightGroup

* Remove HueLightGroup and add is_group variable to HueLight

* Make linter happy

* Update light or lightgroup state when a new state is available

* Use schedule_update_ha_state() to schedule the state update. Drop new_lightgroups and use new_lights instead.

* code style fix

glance- committed:
Improve denon media_player (#4836)

* Add debug level logging of messages in denon

* Added media stop for Denon AVR Media Player

* Sort source list

* Rework input selection for Denon AVR

This reworks the input selection, adding more modes and making it so
that the media controls are only announced in modes where they actually
makes sense.

* Added real media info for Denon AVR Media modes

* Read more configuration from denon devices

This reads network name, and overrides the local name with that.

This also reads the source names and reconfigures the input list to
those names, and also reads the source deleted list and removes the
inputs that are set to deleted in the device.

* Discover and handle max volume in Denon media player

* Rework source discovery in Denon media player

This uses SSFUN as authorative source for which sources that we should
present.

molobrakos committed:
Telldus Live: (#4645)

- Implemented support for covers and dimmable lights.
- Removed global object, use hass.data.
- Disabled polling via update.
- Inherit from common TelldusLiveEntity device.
- Configurable polling interval
- Use https API endpoint
- Use tellduslive package

mezz64 committed:
Add media position support and trailer type to Emby (#4792)

* Add media position support and trailer type to Emby

* Adjustments to mitigate TypeError

* Simplify media_position property

* Update handling when data isn't available

* Update emby.py

emlove committed:
Flic: Support use of queued events within timeout (#4822)

* Flic: Support use of queued events within timeout

* Linter fixes

tchellomello committed:
Added resolution support to Amcrest cameras (#4860)

* Added resolution support to Amcrest cameras

* Ordered alphabetically DEFAULT_ options

balloob committed:
Merge branch 'dev' into master

balloob committed:
Merge pull request #4863 from home-assistant/master

Backmerge point releases

molobrakos committed:
[tellduslive] Upgrade requirement (#4865)

molobrakos committed:
Display error message instead of exception (#4866)

* Display error message instead of exception

Display error message in log instead of stack trace.
(Usually happens when a server is already running at the same port.)

* Update __init__.py

Better error handling when reading SSL certificate

* Update __init__.py

* Update __init__.py

balloob committed:
Fix dev tag detection in release script (#4873)

mezz64 committed:
Add Hikvision binary sensor component (#4825)

* Add Hikvision binary sensor component

* Simplify customize configuration

* Add delay attribute

* Remove use of threading timer, fix delay functionality

pvizeli committed:
TTS Component / Google speech platform (#4837)

* TTS Component / Google speech platform

* Change file backend handling / cache

* Use mimetype / rename Provider function / allow cache on service call

* Add a memcache for faster response

* Add demo platform

* First version of unittest

* Address comments

* improve error handling / address comments

* Add google unittest & check http response code

* Change url param handling

* add test for other language

* Change hash to sha256 for same hash on every os/hardware

* add unittest for receive demo data

* add test for error cases

* Test case load from file to mem over aiohttp server

* Use cache SpeechManager level, address other comments

* Add service for clear cache

* Update service.yaml

* add support for spliting google message

auduny committed:
Remove libtelldus-core-dev from Dockerfile (#4878)

Remove unnecessary  libtelldus-core-dev from Dockerfile . Ref https://github.com/home-assistant/home-assistant/pull/4680#issuecomment-266006310

pvizeli committed:
Migrate REST switch to async (#4517)

* Migrate REST switch to async

* Update rest.py

* Address comments from paulus

molobrakos committed:
Update __init__.py (#4877)

Cleaner exit by not throwing exception if server was not set during initialization of component (ref https://github.com/home-assistant/home-assistant/pull/4866)

scarface-4711 committed:
Added a volume set option and autodiscovery functions to Denon AVR rece… (#4845)

* Added Volume Set option and autodiscovery functions to Denon AVR receivers

* Corrected issues in SSDP discovery and in case no host could be discovered

* Corrected discovery handling / added denonavr to discovery platform

* No needless discoveries anymore / add_devices() with list instead of loop

molobrakos committed:
bugfix: is_on is a property (#4889)

tchellomello committed:
Fixes issues #4844 to avoid traceback when self.rest.data is None (#4886)

6-12-09 18:12:30 homeassistant.core: Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 237, in _step
    result = next(coro)
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/helpers/entity_component.py", line 386, in _update_entity_states
    yield from update_coro
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/helpers/entity.py", line 240, in async_update_ha_state
    self._attr_setter('entity_picture', str, ATTR_ENTITY_PICTURE, attr)
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/helpers/entity.py", line 307, in _attr_setter
    value = getattr(self, name)
  File "/srv/hass/hass_venv/lib/python3.4/site-packages/homeassistant/components/sensor/wunderground.py", line 176, in entity_picture
    url = self.rest.data['icon_url']
TypeError: 'NoneType' object is not subscriptable

Commit:
Update frontend

balloob committed:
Fix hue groups on older hubs (#4884)

Danielhiversen committed:
Bug in rpi_camera

valentinalexeev committed:
[media_player.sonos] Source selection from favorites (#4804)

pvizeli committed:
TTS add google language list for config check (#4912)

* Add config check for language

* update default

* move language from component to platform

* fix lint

Commit:
Update flux led lib

Danielhiversen committed:
Merge pull request #4907 from home-assistant/rpi_camera

Bug in rpi_camera

Danielhiversen committed:
Merge pull request #4924 from home-assistant/flux_lib

Update flux led lib

lwis committed:
[media_player.sonos] Bugfix, initalise source_name. (#4911)

molobrakos commented:
Not sure about the proper procedure here, but I would like to have the bugfix in https://github.com/home-assistant/home-assistant/pull/4917 merged into this release.

pvizeli committed:
Homematic faster update with async (#4929)

pvizeli committed:
Protect add_job (#4932)

joopert committed:
kodi fanart fix basic auth (#4930)

Danielhiversen committed:
Bug fix for #4903 (#4927)

joerocklin committed:
Add nest hvac state (#4810)

* Add basic property details for Nest hvac_state

* Add the hvac_state sensor

* Update requirements and remove trailing whitespace

Clean up the multiline docstring

Adding a space between summary and description

* Removing the hvac_state as a property on the nest climate

* Update nest.py

jabesq committed:
Fix for GTFS sensor (#4828)

* Fix for GTFS sensor

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* GTFS fix

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

Danielhiversen committed:
Broadlink sensor and switch (#4834)

* Broadlink sensor and switch

* broadlink logging

* Use async

* style

* style

trisk committed:
Expose media volume as emulated Hue brightness (#4869)

* Allow virtual Hue bridge to set volume level of media_player entities
* Show correct states in all lights view

magas0 committed:
Panasonic viera fix (#4888)

* Removed return False so the Panasonic Viera TV can be added even if it doesn't connect

* Removed return False so the Panasonic Viera TV can be added even if it doesn't connect

* Removed return False so the Panasonic Viera TV can be added even if it doesn't connect

* Remove try/except to connect to the TV

* Update panasonic_viera.py

* Update panasonic_viera.py

fabaff committed:
Clean-up (#4894)

balloob committed:
Fix synologydsm (#4895)

fabaff committed:
Avoid TypeError for state (#4897)

magicus committed:
Add support for new netdisco detection of Samsung Smart TV. (#4925)

emlove committed:
Setup DarkSky platform when offline during init (#4919)

* Setup DarkSky platform when offline during init

* Fail setup_platform if fetch was unsuccessful

molobrakos committed:
initialize self._last_brightness (#4917)

roidayan committed:
Fix webostv component to accept any custom sources (#4915)

Updated the schema check to accept any string
Search custom sources in app title and app id
The makes the short list redundant and thus removed
Tested by adding livetv, netflix, youtube, makovod and others
This is also compatible with the list that was supported till now
so current users won't see any difference.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

nkgilley committed:
add manual option to prevent scheduled tests. (#4906)

pvizeli committed:
Add voicerss for TTS (#4916)

* Add voicerss for TTS

* add unittests

* fix tests

* fix status bug in google/voicerss

* remove ssl

balloob committed:
Cast fix (#4939)

* Update frontend

* Fix exception on cast startup

Landrash commented:
Call for help for HASSbian looks good 😊

Danielhiversen commented:
For the release notes:

**Vlc media player**
Let you control vlc media player and  play sound on connected speakers.


**Broadlink support**
This Broadlink switch platform allow to you control Broadlink RM2 Pro and RM mini IR+RF devices.
This Broadlink sensor platform adds support for Broadlink RM2 and A1 Devices.


**GpsLogger**
The [GpsLogger](https://home-assistant.io/components/device_tracker.gpslogger/) now supports attributes as the speed, direction, altitude, provider and activity.

trisk committed:
Use Wake-on-LAN to turn on Panasonic Viera TV (#4809)

pvizeli committed:
Bugfix asyncio wait (#4946)

Commit:
[media_player.braviatv] Add turn on capabilities. (#4938)

pvizeli committed:
Async logging file handler (#4901)

* Async logging file handler

* add time rotation handle

* new layout

* address paulus comments

* fix lint

balloob committed:
Update coverage

danielperna84 committed:
Fixing issue #4899 (#4947)

trisk committed:
Use Wake-on-LAN to turn on LG webOS TV (#4808)

fabaff committed:
Upgrade python-telegram-bot to 5.3.0 (#4955)

fabaff committed:
Upgrade astral to 1.3.3 (#4956)

fabaff committed:
Upgrade Sphinx to 1.5.1 (#4957)

molobrakos committed:
eliqonline lib upgrade (#4948)

pvizeli committed:
Bugfix async log handler (#4954)

* Bugfix async log handler

* fix boostrap test

* Use hass.data for store handler and cleanup on async_stop

* Update bootstrap.py

balloob committed:
Only report slowness warning once per entity (#4962)

kirichkov committed:
Updates TP-Link dependency (#4914)

* Updates TP-Link switches dependent module

Refactors code to use the new module API

* Set TP-Link Switch name from the device settings

If no name has been set in the configuration file the name set on the device will be used

* Removes default name for TP-Link switch

Fallback to device alias now works properly

* Removes logging

* Updates comment to denote support for HS200 switch

balloob committed:
Version bump to 0.35.0
